### ISSUE TITLE
crash when playing yt-dlp

### ISSUE THREAD

**Issue by Obegg** [mpv-player/mpv#13918]
### Important Information

Provide following Information:
- mpv version:
```
mpv v0.37.0-793-g32817198 Copyright © 2000-2024 mpv/MPlayer/mplayer2 projects
 built on Apr 16 2024 23:05:35
libplacebo version: v7.349.0 (v6.338.0-121-ge4e096b-dirty)
FFmpeg version: N-114842-g639013aaf
FFmpeg library versions:
   libavutil       59.15.100
   libavcodec      61.5.103
   libavformat     61.3.100
   libswscale      8.2.100
   libavfilter     10.2.101
   libswresample   5.2.100
```
- Windows Version:
```
Edition	Windows 10 IoT Enterprise LTSC
Version	21H2
Installed on	‎10/‎04/‎2024
OS build	19044.4291
Experience	Windows Feature Experience Pack 1000.19056.1000.0
```
- Source of the mpv binary: https://github.com/zhongfly/mpv-winbuild/releases
- If known which version of mpv introduced the problem: Yes.
```
`mpv-x86_64-v3-20240416-git-d23f641` = No issues.
`mpv-x86_64-v3-20240417-git-3281719` = Issues.
```
- GPU model, driver and version:
```
NVIDIA GeForce GTX 1650
Studio Driver 552.22
```
- Possible screenshot or video of visual glitches: No

### Reproduction steps
Download [mpv-x86_64-v3-20240417-git-3281719.7z](https://github.com/zhongfly/mpv-winbuild/releases/download/2024-04-17-3281719/mpv-x86_64-v3-20240417-git-3281719.7z) and [latest yt-dlp](https://github.com/yt-dlp/yt-dlp/releases/tag/2024.04.09)
Open any YouTube video

### Expected behavior
Playing

### Actual behavior
Crash

### Log file
Log file doesn't catch the crash, the log file doesn't show any errors and just stops writing logs once crash has occurred, so log file in this case is impossible to provide.

**Comment by verygoodlee** [comment]
I can also reproduce in shinchiro's build https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20240418

**Comment by verygoodlee** [comment]
yes, zhongfly build mpv-player/mpv@3281719 two times.
- [2024-04-16 23:09](https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-04-16-3281719) works
- [2024-04-17 12:12](https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-04-17-3281719) doesn't work

so it's definitely not a mpv bug.

**Comment by zhongfly** [comment]
> yes, zhongfly build [3281719](https://github.com/mpv-player/mpv/commit/3281719) two times.
> 
> * [2024-04-16 23:09](https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-04-16-3281719) works
> * [2024-04-17 12:12](https://github.com/zhongfly/mpv-winbuild/releases/tag/2024-04-17-3281719) doesn't work
> 
> so it's definitely not a mpv bug.

They have the same ffmpeg source version, both https://github.com/FFmpeg/FFmpeg/commit/639013aafcd9c534c487070c468e2f5309975fe1
Since it's https related, I guess openssl could be the source of the problem: the build that works uses https://github.com/openssl/openssl/commit/c062403abd71550057b3647b01cc8af4cc2fc18c , the build that doesn't work is https://github.com/openssl/openssl/commit/96939f1e2c6ac1264142ef62c9925786f5723649

**Comment by CharlesMengCA** [comment]
LLVM is not happy with the recent code changes.

shinchiro latest build [20240418](https://github.com/shinchiro/mpv-winbuild-cmake/releases/tag/20240418)
 working: https://github.com/shinchiro/mpv-winbuild-cmake/releases/download/20240418/mpv-x86_64-gcc-20240418-git-2f8038c.7z
Not working: https://github.com/shinchiro/mpv-winbuild-cmake/releases/download/20240418/mpv-x86_64-v3-20240418-git-dc632e1.7z

**Comment by kasper93** [comment]
Not a mpv issue.

**Comment by verygoodlee** [comment]
it‘s an issue with openssl, becasue zhongfly's build temporarily fixes the problem zhongfly/mpv-winbuild@789f2ab
